style: apply /style-guide pass to models/runs (#2724)
wandb/docs: models/runs/filter-runs.mdx

---

> description: Learn how to filter runs in the Runs table with the expression editor.

Reviewer: The Runs table and Workspace use the same view/runset filter state.

```suggestion
description: Learn how to filter runs with the expression editor.
```

Author: Applied — shortened the description.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

title: Filter runs
---

> Filter runs based on their name, state, [tags](#example-filter-runs-with-tags), or other properties with the expression editor in the Runs table. Filtering helps you narrow a large set of runs down to the subset you want to compare, analyze, or act on.

Reviewer: ```suggestion
Use the expression editor to filter runs by name, state, [tags](#example-filter-runs-with-tags), or other properties. Filtering narrows a large set of runs to the subset you want to compare, analyze, or act on.

The Runs table and Workspace use the same filters for a view. When you update filters in either place, the same set of runs appears in both the table and workspace visualizations.

<Info>
Runs table filters and Workspace filters use the same underlying runset configuration. Filtering in the Runs table changes which runs are included in the Workspace. Visibility settings, such as the eye toggle, only control whether included runs appear in panels. Some panels might display only a subset of visible runs because of panel-specific limits.
</Info>
```

Author: Applied your full suggestion, including the `<Info>` callout.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

When you add a filter, you first choose a field (for example, tags, timestamp, or entity). Each field has an underlying type, such as text, time, or ID.

The list of operators you see (for example, `is`, `in`, `>=`, `within last`) depends on this type. After you choose a field, the UI shows only operators that are valid for that field's type.

## Common operators by type

The following table shows the operators most commonly used with each field type, along with an example of how each type appears in a filter expression.

| Filter type | Example fields | Common operators | Example usage |
| ----------- | -------------- | ---------------- | ------------- |
| Tags        | `tags`           | `is`, `is not`, `in`, `not in`  | `tags is "baseline"` |
| Time        | `created timestamp`      | `<=`, `>=`, `within last` | `created timestamp >= 2026-01-16` |
| String      | `state`         | `=`, `!=`, `IN`, `NOT IN` | `state = "finished"` |

<Note>

> The preceding table shows only a subset of available fields and operators. The expression editor shows all available fields and operators.

Reviewer: ```suggestion
The previous table shows only a subset of available fields and operators. The expression editor shows all available fields and operators.
```

Author: Applied — "preceding" to "previous".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

</Note>

## Create a filter expression

> Follow these steps to build a filter expression that limits which runs appear in the Runs table.

Reviewer: ```suggestion
The following steps outline how to create a filter:
```

Author: Applied your suggested rewording.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

1. Click the **Runs** tab from the project sidebar.
2. Click the **Filter** button, which looks like a funnel, above the runs table.
3. To create a filter expression, from left to right, select a column name, a logical operator, and a filter value. W&B applies the filter as soon as you select a filter value.
4. Optional: Select **New Filter** to apply additional AND or OR conditions.
5. Optional: Select **New group** to group filters together with parentheses. Grouping lets you create complex filter expressions such as [A AND (B OR C)](#example-combine-filters-with-and-and-or-conditions).
6. To close the filter expression editor, click the **x** icon in the top right corner.

The following image filters runs based on loss values less than or equal to `1`:

<Frame>
    <img src="/images/data_vis/filter.png" alt="Filter runs by loss values less than or equal to 1" />
</Frame>

The following sections show examples of common filter expressions.

### Example: Filter runs with tags

> Use tag-based filters to scope the Runs table to a specific group of runs, such as runs sharing a baseline or experiment label. Filter runs based on their tags:

Reviewer: ```suggestion
Use tag-based filters to scope a specific group of runs, such as runs sharing a baseline or experiment label. Filter runs based on their tags:
```

Author: Applied — dropped "the Runs table to".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

---
description: Learn how to filter runs in the Runs table with the expression editor.
title: Filter runs
keywords: ["filter expression", "runs table", "tags filter", "AND OR conditions"]
---

> Filter runs based on their name, state, [tags](#filter-runs-with-tags), or other properties with the expression editor in the Runs table. Filtering helps you narrow a large set of runs down to the subset you want to compare, analyze, or act on.

Author: Fixed: anchor is now `#example-filter-runs-with-tags` to match the `### Example: Filter runs with tags` heading.

This section describes how to create a complex filter expression like `A AND (B OR C)`. In this example, A is `loss <= 1`, B is `State = finished`, and C is `State = crashed`.

1. Click the **Runs** tab from the project sidebar.
2. Click the **Filter** button, which looks like a funnel, above the runs table.
3. From left to right, select `loss` from the first dropdown menu, select `<=` from the second dropdown menu, and enter `1` in the third dropdown menu. This creates filter A.
4. Click **New Filter** and select **New group** to create a new group for filters B and C.
5. For filter B, select `State` from the first dropdown menu, select `=` from the second dropdown menu, and select `finished` from the third dropdown menu.
6. Click **New Filter** to create filter C, and select `State` from the first dropdown menu, select `=` from the second dropdown menu, and select `crashed` from the third dropdown menu.

The following image demonstrates how the resulting filter expression appears in the filter expression editor:

<Frame>
    <img src="/images/runs/run_expressions_and_or.png" alt="Combine filters with AND and OR conditions" />
</Frame>

## Default filters

> Besides filters you create, W&B includes a few built-in filters that you can toggle on or off to narrow the Runs table without writing an expression. By default, W&B provides the following filters:

Reviewer: ```suggestion
Besides filters you create, W&B includes a few built-in filters that you can toggle on or off to narrow which runs are visable without writing an expression. By default, W&B provides the following filters:
```

Author: Applied your suggestion, with "visable" corrected to "visible".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

1. Click the **Runs** tab from the project sidebar.
2. Click the **Filter** button, which looks like a funnel, above the runs table.
3. From left to right, select `loss` from the first dropdown menu, select `<=` from the second dropdown menu, and enter `1` in the third dropdown menu. This creates filter A.
4. Click **New Filter** and select **New group** to create a new group for filters B and C.
5. For filter B, select `State` from the first dropdown menu, select `=` from the second dropdown menu, and select `finished` from the third dropdown menu.
6. Click **New Filter** to create filter C, and select `State` from the first dropdown menu, select `=` from the second dropdown menu, and select `crashed` from the third dropdown menu.

The following image demonstrates how the resulting filter expression appears in the filter expression editor:

<Frame>
    <img src="/images/runs/run_expressions_and_or.png" alt="Combine filters with AND and OR conditions" />
</Frame>

## Default filters

Besides filters you create, W&B includes a few built-in filters that you can toggle on or off to narrow the Runs table without writing an expression. By default, W&B provides the following filters:

> * **Show only my works**: Shows only runs created by the current user.

Author: Fixed the grammar: "Show only my works" is now "Show only my work". Could not verify the exact live UI label, so worth confirming if it differs.
